Assets/Scripts/KatamariMovement.cs

```diff
@@ -342,22 +309,131 @@ void PowerBar()
         }
 
 
-        power_bar.GetComponent<Image>().fillAmount = power;
+        powerBar.GetComponent<Image>().fillAmount = power;
 
 
     }
 
     // Returns True if in golf hit mode
     public bool IsGolfHitMode()
     {
+
+        if (myStateMachine == stateMachine.golfMode)
+        {
+            isGolfHitMode = true;
+        } else
+        {
+            isGolfHitMode = false;
+        }
         return isGolfHitMode;
     }
 
+    public void time_state_machine() {
+        // THINGS TO CHANGE: Replace canhit() with end of stoke condition
+       
+
+        bool isFlyMovementDeadTime = resourceManager.GetTimeSinceLastHit() >= 0.75f;
+        
+        //GOLF MODE\\
+        if (myStateMachine == stateMachine.golfMode)
+        {
+            
+            _PowerBar();
+            strikeStrength = power * 2500; // make this editable
+
+            //Transition to Normal Speed\\
+            if (hitInput >0.5){
+                
+                myStateMachine = stateMachine.normalSpeed;
+                powerBar.SetActive(false);
+                this.GetComponent<LineRenderer>().enabled = false;
+                resourceManager.tryToHit();
+                forceMode = ForceMode.Force;
+                
+            }
+        }
+
+        if (myStateMachine == stateMachine.toGolfMode) {
+            Time.timeScale += (1f / slowdownLength) * Time.unscaledDeltaTime;
+            Time.timeScale = Mathf.Clamp(Time.timeScale, slowdownFactor, 1f);
+            Time.fixedDeltaTime = Time.timeScale * 0.02f;
```

Comment:
I think the `Time` logic is duplicated here and in `TimeManager`. Naming-wise it could make sense to leave it in `TimeManager`.

Reply:
oh yeah we dont need TimeManager at all anymore!